File: 0299-bulls-and-cows/0299-bulls-and-cows.js
/**
 * @param {string} secret
 * @param {string} guess
 * @return {string}
 */
var getHint = function(secret, guess) {
  const count = new Array(10).fill(0);
  let [a, b] = [0, 0];
  
  for (let i = 0; i < secret.length; i++) {
    const [s, g] = [secret[i], guess[i]];
    if (s !== g) {
      if (count[g]++ < 0) b++;
      if (count[s]-- > 0) b++;
    }
    
    else a++;
  }
  
  return `${a}A${b}B`;
};

File: 0299-bulls-and-cows/0299-bulls-and-cows.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        count = Counter()
        a, b = 0, 0
        for s, g in zip(secret, guess):
            if s == g:
                a += 1
            else:
                if count[g] < 0:
                    b += 1
                count[g] += 1
                if count[s] > 0:
                    b += 1
                count[s] -= 1
        return f"{a}A{b}B"